src/ET_Client.php

```diff
@@ -165,6 +178,18 @@ function __construct($getWSDL = false, $debug = false, $params = null)
                     $this->baseSoapUrl = $this->defaultBaseSoapUrl;
 				}
 			}
+			if ($params && array_key_exists('useOAuth2Authentication', $params))
```

Comment:
can we simplify these if loops?

Reply:
We can remove the first expression - `$params` - from the inner `if`s, since the outer `if` already does the same test.